lib/cc/analyzer/issue_validator.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Analyzer
+    class IssueValidator
+      CHECKS = [
+        IssueCategoryValidation,
+        IssueCheckNamePresenceValidation,
+        IssueDescriptionPresenceValidation,
+        IssuePathExistenceValidation,
+        IssuePathPresenceValidation,
+        IssueRelativePathValidation,
+        IssueTypeValidation,
+      ].freeze
+
+      attr_reader :error
+
+      def initialize(issue)
+        @issue = issue
+      end
+
+      def valid?
+        @error.blank?
+      end
+
+      def validate
+        if issue && invalid_messages.any?
+          @error = {
+            message: "#{invalid_messages.join("; ")}: `#{issue}`.",
+            issue: issue,
+          }
+          false
+        else
+          true
```

Comment:
Yeah the unparseable case is handled [elsewhere](https://github.com/codeclimate/codeclimate/blob/gd-issue-validation/lib/cc/analyzer/issue.rb#L50) (we let JSON.parse raise in this repo). In builder we handle it separately. Because it's a different error code (E12 vs E13), I think not having a `Parseable JSON` validation is better